src/components/ProfileCard/ProfileCard.tsx

```diff
@@ -0,0 +1,161 @@
+import * as React from 'react';
+
+import {
+  DEFAULT_PROPS,
+  STATS_ICONS,
+  STATS_LABELS,
+  DEFAULT_AVATAR_ICON,
+  FALLBACK_INITIALS,
+} from './ProfileCard.constants';
+import {
+  StyledCard,
+  StyledProfileContainer,
+  StyledAvatar,
+  StyledInfoContainer,
+  StyledName,
+  StyledUsername,
+  StyledRole,
+  StyledStatsContainer,
+  StyledStatItem,
+  StyledStatIcon,
+  StyledStatContent,
+  StyledStatValue,
+  StyledStatLabel,
+} from './ProfileCard.styles';
+
+import type { ProfileCardProps, StatsItemProps } from './ProfileCard.types';
+
+const StatsItem: React.FC<StatsItemProps> = ({ icon, label, value, 'data-testid': dataTestId }) => {
+  const formattedValue = typeof value === 'number' ? value.toLocaleString('en-US') : value;
+
+  return (
+    <StyledStatItem $variant={DEFAULT_PROPS.variant} data-testid={dataTestId}>
+      <StyledStatIcon>{icon}</StyledStatIcon>
+      <StyledStatContent $variant={DEFAULT_PROPS.variant}>
+        <StyledStatValue>{formattedValue}</StyledStatValue>
+        <StyledStatLabel>{label}</StyledStatLabel>
+      </StyledStatContent>
+    </StyledStatItem>
+  );
+};
+
+/** Profile card component for displaying user information and statistics */
+export const ProfileCard: React.FC<ProfileCardProps> = ({
+  avatarUrl,
+  name,
+  username,
+  role,
+  stats = { courses: 0, points: 0, ranking: 0 },
+  size = DEFAULT_PROPS.size,
+  variant = DEFAULT_PROPS.variant,
+  showStats = DEFAULT_PROPS.showStats,
+  className,
+  'data-testid': dataTestId = 'profile-card',
+  ...rest
+}) => {
+  // Strict validation for required props
+  if (!name || !username) {
+    throw new Error('ProfileCard: "name" and "username" are required props and cannot be empty.');
+  }
+
+  const DefaultIcon = DEFAULT_AVATAR_ICON;
+  const initials = FALLBACK_INITIALS(name);
+
+  const formatRanking = (rank: number): string => {
+    if (rank === 0) return '-';
+
+    const lastDigit = rank % 10;
+    const lastTwoDigits = rank % 100;
+
+    // Special cases for 11, 12, 13 (always 'th')
+    if (lastTwoDigits >= 11 && lastTwoDigits <= 13) {
+      return `${rank}th`;
+    }
+
+    // Regular cases based on last digit
+    if (lastDigit === 1) return `${rank}st`;
+    if (lastDigit === 2) return `${rank}nd`;
+    if (lastDigit === 3) return `${rank}rd`;
+
+    return `${rank}th`;
+  };
+
+  const safeStats = stats || DEFAULT_PROPS.stats;
+
+  return (
+    <StyledCard
+      {...rest}
+      className={className}
+      data-testid={dataTestId}
+      $size={size}
+      $variant={variant}
+      role="article"
```

Comment:
The role and region attributes are set (e.g. role="article"). Ensure that `role` isn't accidentally duplicated via `...rest` (e.g. if caller passes a role prop) which could cause duplicate attributes or unexpected overrides.

**Sugerencia:**
```typescript
Either avoid spreading all `...rest` to the root and explicitly pick allowed CardProps, or ensure `role` and ARIA attributes are overridden intentionally (for example prefer explicit prop precedence: `{...rest}` after explicit attributes or validate `rest.role`). Example:

const { role: _role, ...cardRest } = rest;
<StyledCard {...cardRest} role="article" ...>
```